third_party/2and3/markdown/preprocessors.pyi

```diff
@@ -1,14 +1,14 @@
-from typing import Any, Pattern
+from typing import Any, List, Pattern
 
 from . import util
 
 def build_preprocessors(md, **kwargs): ...
 
 class Preprocessor(util.Processor):
-    def run(self, lines) -> None: ...
+    def run(self, lines: List[str]) -> List[str]: ...
 
 class NormalizeWhitespace(Preprocessor):
-    def run(self, lines): ...
+    def run(self, lines: List[str]) -> List[str]: ...
```

Comment:
While having a tight type in the (loosely) abstract `Preprocessor` class makes sense, I suggest to use `Iterable` here in the implementation. Same below.

Reply:
str.split returns a List, so I believe this is the best type.

Reply:
I meant the argument type, sorry for being ambiguous.

Reply:
Oh right, that makes way more sense

Reply:
```suggestion
    def run(self, lines: Iterable[str]) -> List[str]: ...
```